builder/builder.go

```diff
@@ -63,8 +63,10 @@ func (b *Builder) RunTask(ctx context.Context, task *graph.Task) error {
 		timeout := time.Duration(loginTimeoutInSec) * time.Second
 		loginCtx, cancel := context.WithTimeout(ctx, timeout)
 		defer cancel()
-		if err := b.dockerLoginWithRetries(loginCtx, task.RegistryName, task.RegistryUsername, task.RegistryPassword, 0); err != nil {
-			return err
+		for _, creds := range task.Credentials {
```

Comment:
Do you want to login in parallel (max of 3 in parallel) just to make sure that the overall execution time doesn't increase? Or do you want to optimize that as another item /cc @northtyphoon

Reply:
Synced with @ehotinger , I will add an enhancement to make it parallel before closing this PR. For now, it won't affect perf a lot because there won't be many credentials